### DISCUSSION TITLE
inter-occasion variability rxode2

### DISCUSSION THREAD

**Discussion by jareberg** [nlmixr2/rxode2#322] [Q&A]
Hi,

I'm trying to incorporate iov in an rxode2 model using the instructions for RxODE. My code looks like

```r
model<-RxODE({
  KA<-KA*exp(KA_eta)
  V<-V*exp(V_eta)
  CL<-CL*exp(CL_eta+iov.cl)
  cp=A2/V*(1+eps)
  d/dt(A1)<- -KA*A1;
  d/dt(A2)<- KA*A1-CL/V*A2;
})

#Create an eventTable
ev<-eventTable(amount.units = 'ug',time.units = "hours")
ev$add.dosing(dose=100000,nbr.doses = 1,dosing.to = 1,start.time = 0)
ev$add.dosing(dose=100000,nbr.doses = 1,dosing.to = 1,start.time = 100)
ev$add.sampling(0:200)
ev$occ<-ifelse(ev$time<100,1,2)

#Give initial conditions
init<-c(0,0) # Give starting values for each compartment

#Parameter values
params<-c(KA=1,CL=10,V=100)

#Omega matrix
omega<-lotri(lotri(V_eta+CL_eta~
               c(0.1,
                 0.01,0.1))|id(nu=100),
             lotri(iov.cl~0.1)|occ(nu=200))
#Sigma matrix
sigma<-lotri(eps~0.2)

#Run the simulation
sim<-rxSolve(model,events=ev,inits=init,params=params,
           sigma=sigma,omega=omega,nSub=100)
```

However, when I try to simulate on the model 
_sim<-rxSolve(model,events=ev,inits=init,params=params,
           sigma=sigma,omega=omega,nSub=100)_
I get following error message
Error in rxSolveSEXP(object, .ctl, .nms, .xtra, params, events, inits,  : 
  Not compatible with STRSXP: [type=NULL].

Am I doing something wrong or has iov implementation been changed for rxode2 ?

Best regards
Johan Areberg

**Comment by mattfidler** [comment]
I believe this is a bug.  I will see if I what I can do here.

**Reply by mattfidler** [reply]
In the mean time, you can code in the nonmem way by adding multiple etas and then indicating if that eta should be included.
